app/src/main/java/com/example/notes2text/fileselection/adapters/SelectionController.java

```diff
@@ -0,0 +1,183 @@
+
+        // Retrieve the file path from the intent.
+//        String path = this.getArguments().getString("path");
+        //Use the filepath local variable as source.
+        // Get the address of the folder currently to be shown.
+        File root = new File(filePath);
+        // Retrieve the list of files and folders from the path address.
+        File[] filesDirectory = root.listFiles();
+
+        TextView noFiles = view.findViewById(R.id.noFilesText);
+        RecyclerView fileListView = view.findViewById(R.id.file_list);
+        //Depending on the state of the list of files, display either No Files text or the list
+        // of files as view holder objects.
+        if (filesDirectory == null || filesDirectory.length == 0){
+            noFiles.setVisibility(View.VISIBLE);
+        } else{
+            // set noFiles to Invisible
+            noFiles.setVisibility(View.INVISIBLE);
+            //Assign Linear layout to file list.
+            fileListView.setLayoutManager(new LinearLayoutManager(this.getActivity()));
+            //Assign the custom adaptor to the View elements.
+            if (fileList != null & fileList.isEmpty()){
+                fileListView.setAdapter(new SelectionListAdapter(getActivity().getApplicationContext(), filesDirectory, selectionUseCase, fileList));
+            } else if (fileList != null ) {
+                fileListView.setAdapter(new SelectionListAdapter(getActivity().getApplicationContext(), filesDirectory, selectionUseCase, fileList));
+                selectionPresenter.InheritFilesSuccess(getActivity());
+            } else{
+                fileListView.setAdapter(new SelectionListAdapter(getActivity().getApplicationContext(), filesDirectory, selectionUseCase));
+            }
+        }
+
+        //Add top menu and on click listener for top menu buttons.
+        // Put functionality of top menu items in separate use case.
+        // select listener goes in this class.
+//        Toolbar directoryToolbar = view.findViewById(R.id.file_manager_toolbar);
+//
+//        ((AppCompatActivity) getActivity()).setSupportActionBar(directoryToolbar);
+
+        //call method to
+        setFragmentToolbar(view);
+
+        setToolbarMenu();
+
+
+    }
+
+    //helper method to set toolbar menu.
+    private void setFragmentToolbar(View view) throws NullPointerException {
+        Toolbar selectionToolbar = view.findViewById(R.id.selection_manager_toolbar);
+        ((AppCompatActivity) requireActivity()).setSupportActionBar(selectionToolbar);
+    }
+
+    private void setToolbarMenu() {
+        requireActivity().addMenuProvider(new MenuProvider() {
+            @Override
+            public void onCreateMenu(@NonNull Menu menu, @NonNull MenuInflater menuInflater) {
+                //Inflate the menu
+                menuInflater.inflate(R.menu.selection_toolbar, menu);
+            }
+
+            @Override
+            public boolean onMenuItemSelected(@NonNull MenuItem menuItem) {
```

Comment:
The return value seems to be unused, perhaps there are other usage of it or more suitable methods?